scico/solver.py

```diff
@@ -577,3 +577,371 @@ def golden(
     else:
         r = x
     return r
+
+
+class SolveATAD:
```

Comment:
Consider using a noun for the class name: e.g., `MatrixSystemSolver` or `ATADSolver`.

Reply:
Good point. Done.